### PR TITLE
deprecation: fix createVideoThumbnail

### PR DESCRIPTION
**Pull request by david-allison-1**
## Purpose / Description
I saw an exception in the logs when playing Audio cards, and wanted to avoid this

Now we display `E/MediaMetadataRetrieverJNI: getEmbeddedPicture: Call to getEmbeddedPicture failed`

## Fixes
Related: #5022

## Approach
* Use the new API
* Convert string to a path
* Convert `MINI_SIZE` to  sample representation (method now accepts a `Size`)
* Don't use a cancelListener

## How Has This Been Tested?

Android 11 device: no longer has a large stack trace in logs

## Learning (optional, can help others)

https://developer.android.com/reference/android/media/ThumbnailUtils#createVideoThumbnail(java.io.File,%20android.util.Size,%20android.os.CancellationSignal)
MINI_SIZE: https://github.com/aosp-mirror/platform_packages_providers_mediaprovider/blob/b430fcafc434c559324732b4296f1146b2f717f8/apex/framework/java/android/provider/MediaStore.java#L1697

## Checklist
- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by david-allison-1** [comment]
@Arthur-Milchior Just flagging that this exposes a https://developer.android.com/reference/android/os/CancellationSignal and we may want to consider the implementation of this in our Async framework changes

File: AnkiDroid/src/main/java/com/ichi2/compat/CompatHelper.java [hunk 1]
```diff
@@ -26,8 +26,9 @@
 
 
     private CompatHelper() {
-
-        if (getSdkVersion() >= Build.VERSION_CODES.O) {
+        if (getSdkVersion() >= Build.VERSION_CODES.Q) {
+            mCompat = new CompatV29();
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/compat/CompatHelper.java:30]: How time flies, our first V29 compat API

**Review by mikehardy — APPROVED** [review]
Works for me. Fancy with your Kotlin ;-)

**Comment by david-allison-1** [comment]
Really not a nice class to display Kotlin's new features, but it's overall progress in the right direction (last Kotlin conversion saved 100 lines of code)

**Comment by Arthur-Milchior** [comment]
I beg your pardon, but I don't understand what you want us to consider.
I guess `CollectionTask` could extends `CancellationSignal.OnCancelListener` instead of `CancelListener`, and methods that takes a `CancelListener` could take a `CancellationSignal` instead, in order to separate the signal from the listener. If it's a best practice, it seems simple to implement right now, but I don't see what is the advantage here

**Comment by david-allison-1** [comment]
It would be worth considering the design decisions that Google made with the class in case we differ from them. Partially from a compatibility and partially from a design perspective.

**Comment by Arthur-Milchior** [comment]
There is little documentation on the class, so hard to seriously consider it.
As far as I can tell, the main difference is that you can send the cancel signal before the listener exists, and in this case the listener will receive the signal as soon as it is attached to the signal. You can also disconnect the listener from the signal. Those are nice extra option. I don't really see the need for it right now, but we can add it, and maybe at some point a user will use it since it'll be present. I'm totally fine with it, mostly indifferent

**Pull request merged**